File: frontend/src/types/dsn.ts
export interface DSNSignalInfo {
  active: boolean;
  band: string;
  data_rate_bps: number;
  power_dbm: number;
  signal_type: string;
}

export interface DSNDish {
  dish_name: string;
  complex_id: string;
  complex_name: string;
  location: string;
  flag: string;
  azimuth_deg: number;
  elevation_deg: number;
  rtlt_sec: number;
  range_km: number;
  activity: string;
  down_signal: DSNSignalInfo | null;
  up_signal: DSNSignalInfo | null;
  has_active_signal: boolean;
}

export interface ComplexStatus {
  complex_id: string;
  complex_name: string;
  flag: string;
  is_tracking: boolean;
  dish_count: number;
}

export interface DSNStatus {
  is_tracking: boolean;
  dishes: DSNDish[];
  primary_dish: DSNDish | null;
  complexes: ComplexStatus[];
  fetched_at: string;
  dsn_timestamp_ms: number | null;
}

File: backend/app/models/dsn.py
from pydantic import BaseModel
from typing import Optional, List


class DSNSignalInfo(BaseModel):
    active: bool
    band: str
    data_rate_bps: float
    power_dbm: float
    signal_type: str


class DSNDish(BaseModel):
    dish_name: str
    complex_id: str       # "gdscc" | "mdscc" | "cdscc"
    complex_name: str     # "Goldstone" | "Madrid" | "Canberra"
    location: str
    flag: str
    azimuth_deg: float
    elevation_deg: float
    rtlt_sec: float
    range_km: float
    activity: str
    down_signal: Optional[DSNSignalInfo] = None
    up_signal: Optional[DSNSignalInfo] = None
    has_active_signal: bool


class ComplexStatus(BaseModel):
    complex_id: str
    complex_name: str
    flag: str
    is_tracking: bool
    dish_count: int        # number of dishes on EM2 at this complex


class DSNStatus(BaseModel):
    is_tracking: bool
    dishes: List[DSNDish]
    primary_dish: Optional[DSNDish] = None
    complexes: List[ComplexStatus]
    fetched_at: str
    dsn_timestamp_ms: Optional[int] = None
